pkg/crd/markers/patch_validation.go

```diff
@@ -64,3 +69,22 @@ func (m FeatureSetEnum) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
 	schema.Enum = vals
 	return nil
 }
+
+type FeatureSetXValidation struct {
+	FeatureSetNames []string `marker:"featureSet"`
+	Rule            string
+	Message         string `marker:",optional"`
+}
+
+func (m FeatureSetXValidation) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
+	if !RequiredFeatureSets.HasAny(m.FeatureSetNames...) {
```

Comment:
I am still not so familiar with the code However, this is basically "just" a conditional saying if any of the featureset names is present, treat the validation rule as any other validation rule that's not conditional.

Reply:
Precisely, if we can see that we've been requested to only produce a certain feature set, and that is the feature set for this rule, then we fallthrough to the normal generation pattern